### ISSUE TITLE
Issue for first pagination-labels in  Multiple Instances

### ISSUE THREAD

**Issue by Priyakshi0703** [michaelbromley/ngx-pagination#277]
```html
<div *ngFor="let challenge of entries; let i=index;">
    <mat-expansion-panel  [expanded]="i===0?true:false">
      <mat-expansion-panel-header>
        <div class="leaderboard-useful-points">
          <ul>
            <!-- *ngFor="let subChallenge of item" -->
            <li><i class="icon-holder"><img src="assets/images/lb_icon1.png"></i>{{challenge.challengeName}}</li>
            <li *ngIf="!challenge.creditsAvailable; else somethingElse"><i class="icon-holder"><img src="assets/images/lb_icon2.png"></i>Credits available: N/A</li>
            <ng-template #somethingElse>
              <li *ngIf="challenge.creditsAvailable"><i class="icon-holder"><img src="assets/images/lb_icon2.png"></i>Credits available: {{challenge.creditsAvailable}}</li>
            </ng-template>
            <li><i class="icon-holder"><img src="assets/images/lb_icon3.png"></i>Deadline: {{challenge.deadLine | date}}</li>
          </ul>
        </div>
      </mat-expansion-panel-header>
      <div class="table-responsive">
        <table class="table tableBodyScroll challenge-tableBodyScroll">
          <thead>
            <tr>
              <th *ngFor="let column of challenge.entries.content[0] | columns;">
                {{ column.columnName | titlecase }}</th>
            </tr>
          </thead>
          <!-- <tbody attr.id="table-container{{i}}"> -->
          <tbody attr.id="table-container{{i}}">
            <tr *ngFor="let item of challenge.entries.content | paginate: { itemsPerPage: size, currentPage: page[i] ? page[i] : 1, id:i, totalItems: challenge.entries.totalElements}; let indx=index">
              <!-- <td *ngFor="let column of item | columns">{{ item[column.key] }}</td> -->
              <td *ngFor="let column of item | columns">{{ column.value }}</td> 
            </tr>
            <tr class="no-record" *ngIf="challenge.entries.content.length===0">
              <td>No Record Found.</td>
            </tr>
          </tbody>
          <tfoot *ngIf="challenge.entries.content.length">
            <!-- <tr>
              <td class="show-more" colspan="5"><a href="javascript:void(0)" (click)="getMoreLeaderboards(i, challenge.id, $event)">Show More</a></td>
            </tr> -->
            <pagination-controls maxSize="5" [id]="i" (pageChange)="getMoreLeaderboards($event, i, challenge.id)">
            </pagination-controls>
          </tfoot>
        </table>
      </div>
```

![image](https://user-images.githubusercontent.com/33216636/46414842-a0d64a00-c741-11e8-85b6-cc32a744ec58.png)

upon clicking next button gets this warning

PaginationControlsDirective: the specified id "0" does not match any registered PaginationInstance

**Comment by michaelbromley** [comment]
Hi,

Thanks for the report. Looks like you found a bug! The reason this breaks if due to this line:

https://github.com/michaelbromley/ngx-pagination/blob/bce23596bf329ed4620d3c20b2cf430c12112f63/src/paginate.pipe.ts#L84

In the first row of your `*ngFor`, the i variable is `0`, which of course evaluates to false in the expression above. I just need to change that to an "undefined" check. I'll get a patch out for this shortly.

ps I took the liberty of formatting your code in the issue above. You just need to use three backticks plus the language, e.g.: ` ```html ` at the start of the block and three backticks at the end.

**Comment by michaelbromley** [comment]
Available now as v3.2.1 on npm! 👍

**Comment by Priyakshi0703** [comment]
Thank you
